src/main/java/at/petrak/hexcasting/api/casting/eval/vm/SpellContinuation.kt

```diff
@@ -20,42 +13,38 @@ sealed interface SpellContinuation {
     fun pushFrame(frame: ContinuationFrame): SpellContinuation = NotDone(frame, this)
 
     companion object {
-        // TODO port: maybe serialize to list like before?
-        // TODO port: maybe unit should be first
-        @JvmStatic
-        val CODEC = Codec.recursive<SpellContinuation>(
-            SpellContinuation::class.java.simpleName
-        ) { recursed: Codec<SpellContinuation> ->
-            Codec.withAlternative<SpellContinuation>(
-                Codec.unit(Done),
-                RecordCodecBuilder.create<NotDone> { inst ->
-                    inst.group(
-                        ContinuationFrame.Type.TYPED_CODEC.fieldOf("frame").forGetter { it.frame },
-                        recursed.fieldOf("next").forGetter { it.next }
-                    ).apply(inst, ::NotDone)
-                }
-            )
-        }
-        @JvmStatic
-        val STREAM_CODEC = StreamCodec.recursive<RegistryFriendlyByteBuf, SpellContinuation> { recursed ->
-            withAlternative(
-                StreamCodec.unit(Done),
-                StreamCodec.composite(
-                    ContinuationFrame.Type.TYPED_STREAM_CODEC, NotDone::frame,
-                    recursed, NotDone::next,
-                    ::NotDone
-                ).map(Function.identity()) { it as NotDone }
-            )
+        fun fromList(frames: List<ContinuationFrame>): SpellContinuation {
+            var accumulator: SpellContinuation = Done
+            val it = frames.listIterator(frames.size)
+            while (it.hasPrevious()) {
+                accumulator = NotDone(it.previous(), accumulator)
+            }
+            return accumulator
         }
 
-        private fun <B: ByteBuf, T> withAlternative(primary: StreamCodec<B, T>, alternative: StreamCodec<B, T>): StreamCodec<B, T> {
-            return ByteBufCodecs.either<B, T, T>(
-                primary,
-                alternative
-            ).map<T>(
-                Function { either: Either<T, T> -> Either.unwrap(either) },
-                Function { value: T -> Either.left(value) }
-            )
+        fun toList(continuation: SpellContinuation): List<ContinuationFrame> {
+            val accumulator = ArrayList<ContinuationFrame>()
+            var c = continuation
+            while (c != Done) {
+                accumulator.add((c as NotDone).frame)
```

Comment:
(Nitpick)
Check `is NotDone` like in pre-1.21 hexcasting would allow to avoid casting here
https://github.com/FallingColors/HexMod/blob/5da1bf82464d82aa303020e3fdf1fca17b9055d1/Common/src/main/java/at/petrak/hexcasting/api/casting/eval/vm/SpellContinuation.kt#L25-L26
Not really critical, so eh, replace if you want, or just mark as resolved

Reply:
I explicitly chose `c != Done` because it _should_ just resolve to a reference check rather than a class check (`Done` being an `object`). `is NotDone` is probably better for future readability so I'll discuss it with Sirius